core/src/plugins/filed/python/vmware/BareosFdPluginVMware.py

```diff
@@ -1116,6 +1268,326 @@ def _get_dcftree(self, dcf, folder, vm_folder):
+        """
+        resource_pools = {}
+        resource_pools["/"] = cluster.resourcePool
+        stack = []
+        for resource_pool in cluster.resourcePool.resourcePool:
+            stack.append(("/", resource_pool))
+        while stack:
+            folder, resource_pool = stack.pop()
+            path = "/" + (folder + "/").lstrip("/") + resource_pool.name
+            resource_pools[path] = resource_pool
+            for sub_resource_pool in resource_pool.resourcePool:
+                stack.append((path, sub_resource_pool))
+
+        return resource_pools
+
+    def _get_vm_folder_path(self, vm):
+        """
+        Get the folder path of a given vm
+        """
+        current_obj = vm
+        vm_folder_path = ""
+        while current_obj.parent:
+            current_obj = current_obj.parent
+            if isinstance(current_obj.parent, vim.Datacenter):
+                break
+            if (
+                hasattr(current_obj, "childType")
+                and "VirtualMachine" in current_obj.childType
+            ):
+                vm_folder_path = "/" + current_obj.name + vm_folder_path
+
+        vm_folder_path += "/" + vm.name
+        return vm_folder_path
+
+    def get_resource_pool_by_path(self, resource_pool_path, cluster):
+        """
+        Get resource pool by path
+
+        As resource pools can be nested, unique specfication requires a path-like
+        parameter.
+        """
+        resource_pools = self._get_resource_pools(cluster)
+        if resource_pool_path not in resource_pools:
+            bareosfd.JobMessage(
+                bareosfd.M_FATAL,
+                "Resource pool %s not found in cluster %s!\n"
+                % (
+                    StringCodec.encode(resource_pool_path),
+                    StringCodec.encode(cluster.name),
+                ),
+            )
+            return False
+
+        return resource_pools[resource_pool_path]
+
+    def create_vm(self):
+        """
+        Create a new VM using JSON configuration data
+        """
+
+        destination_host = None
+        resource_pool = None
+        datacenter = None
+        create_vm_task = None
+        cluster_path = None
+        datastore_name = None
+        vm_created = False
+
+        config_info = json.loads(self.restore_vm_config_json)
+
+        if self.options.get("restore_datastore"):
+            datastore_name = self.options["restore_datastore"]
+
+        # prevent from MAC address conflicts
+        self.check_mac_address(config_info)
+
+        # prevent from duplicate UUID
+        self.check_uuid(config_info)
+
+        transformer = BareosVmConfigInfoToSpec(config_info)
+        used_datastore_names = transformer.get_datastore_names()
+        if len(used_datastore_names) > 1:
+            bareosfd.JobMessage(
+                bareosfd.M_FATAL,
+                "VM %s used multiple datastores (%s), restore not yet possible!\n"
```

Comment:
Is the fact that the vm uses multiple datastores already detected during backup? Probably it would then make sense to warn that the backup is not restorable.

Reply:
Good point, added a check at backup time so that it will result with "OK -- with warnings" in that case.